frontend_tests/node_tests/stream_data.js

```diff
@@ -128,6 +135,58 @@ test("basics", () => {
     assert.equal(stream_data.slug_to_name("99whatever"), "99whatever");
 });
 
+test("get_subscribed_streams_of_user", () => {
+    const denmark = {
+        subscribed: true,
+        color: "blue",
+        name: "Denmark",
+        stream_id: 1,
+        is_muted: true,
+        invite_only: true,
+        history_public_to_subscribers: true,
+    };
+    const social = {
+        color: "red",
+        name: "social",
+        stream_id: 2,
+        is_muted: false,
+        invite_only: false,
+        history_public_to_subscribers: false,
+        stream_post_policy: stream_data.stream_post_policy_values.admins.code,
+    };
+    const test = {
+        color: "yellow",
+        name: "test",
+        stream_id: 3,
+        is_muted: true,
+        invite_only: true,
+    };
+    const subs = [denmark, social, test];
+    for (const sub of subs) {
+        stream_data.add_sub(sub);
+    }
+
+    peer_data.set_subscribers(denmark.stream_id, [me.user_id, test_user.user_id]);
+    peer_data.set_subscribers(social.stream_id, [test_user.user_id]);
+    peer_data.set_subscribers(test.stream_id, [test_user.user_id]);
+
+    blueslip.expect(
+        "warn",
+        "We got a is_user_subscribed call for a non-existent or inaccessible stream.",
```

Comment:
We don't want to have normal code that triggers `blueslip` errors; we should be pre-checking whether we have access first.  I'll just change this while I'm reading the code.